source/flux/parts/sis.js

```diff
@@ -78,7 +79,7 @@ export function updateCourseData() {
 
 type CourseDataAction =
 	| LoadCachedCoursesAction
```

Comment:
`LoadCachedCoursesAction` doesn't belong in `CourseDataAction` – well, see [1] – because `CourseDataAction` exists to encapsulate the start/success/failure states of the `TermsUpdate*Action`s.

LoadCachedCoursesAction should be joined into the `type Action` list of types, instead.

[1]: `CourseDataAction` should probably be renamed to `TermsUpdateAction`, to better identify that it encapsulates both `TermsUpdate*Action`s

Reply:
But since updateCourseData() might dispatch LOAD_CACHED_COURSES too, doesn't it's return type need to include the LoadCachedCoursesActionType? That was why I was combining them into CourseDataAction..

Reply:
I think updateCourseData's return type should be a union of `ThunkAction<TermsUpdateAction> | ThunkAction<LoadCachedCoursesActionType>`

Reply:
When I do this I get this error:
```
Error: source/flux/parts/sis.js:67
     	       v------------------
 67: 	return async dispatch => {
 68: 		dispatch({type: TERMS_UPDATE_START})
 69: 		let updateNeeded = await updateStoredCourses()
...:
 75: 	}
     	^ async function. Could not decide which case to select
 66: export function updateCourseData(): UpdateCourseDataActionType {
                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^ union type
  Case 1 may work:
   65: export type UpdateCourseDataActionType = ThunkAction<TermsUpdateAction> | ThunkAction<LoadCachedCoursesAction>
                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ThunkAction
  But if it doesn't, case 2 looks promising too:
   65: export type UpdateCourseDataActionType = ThunkAction<TermsUpdateAction> | ThunkAction<LoadCachedCoursesAction>
                                                                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ThunkAction
  Please provide additional annotation(s) to determine whether case 1 works (or consider merging it with case 2):
   67: 	return async dispatch => {
       	             ^^^^^^^^ dispatch
````

Reply:
Ah. Try `ThunkAction<TermsUpdateAction | LoadCachedCoursesActionType>` instead.